eox_nelp/processors/xapi/mixins.py

```diff
@@ -0,0 +1,47 @@
+"""Mixin file, this is a collection of classes that could be reused in multiple implementations.
+
+Classes:
+    - BaseCourseObjectMixin: Base mixin for transformers that has as object an edx-platform course
+"""
+from django.utils.functional import cached_property
+from tincan import Activity, ActivityDefinition, LanguageMap
+
+from eox_nelp.processors.xapi import constants as eox_nelp_constants
+from eox_nelp.utils import get_course_from_id
+
+
+class BaseCourseObjectMixin:
+    """
+    Base mixin for transformers that has as object an edx-platform course.
+    """
+    @cached_property
+    def course_id(self):
+        """The course id data depends on the kind of event, an event could publish this as course_id or course_key
+        so this a wrapper that should be implemented in order to use the get_object method.
+
+        Raises:
+            NotImplementedError: when the subclass has not overridden the course_id property.
+        """
+        raise NotImplementedError("The property course_id has not been implemented, please set this porperty")
+
+    def get_object(self):
```

Comment:
This gonna be the standard. Anyway, could be overriden.

Reply:
yes, this is the nelc standard for course objects and comes from the [nelc guide](https://docs.google.com/document/d/1T6vPWps8TxDs_M9QAOJBSiDjbTou2iwW9yj66pdiwR0/edit)